aodndata/soop/soop_tmv_nrt.py

```diff
@@ -0,0 +1,154 @@
+    df.replace(np.nan, template.variables['LATITUDE']['_FillValue'], inplace=True)
+
+    template.variables['TIME']['_data'] = time_val_dateobj
+    template.variables['LATITUDE']['_data'] = df.LATITUDE.values
+    template.variables['LONGITUDE']['_data'] = df.LONGITUDE.values
+
+    template.variables['TEMP']['_data'] = df.TEMP.values
+    template.variables['PSAL']['_data'] = df.PSAL.values
+    template.variables['TURB']['_data'] = df.TURB.values
+    template.variables['CPHL']['_data'] = df.CPHL.values
+
+    if '1SecRaw' in log_filename:
+        time_period = '1sec'
+    else:
+        time_period = '10secs'
+
+    template.global_attributes.update({
+        'time_coverage_start': df.index.strftime('%Y-%m-%dT%H:%M:%SZ')[0],
+        'time_coverage_end': df.index.strftime('%Y-%m-%dT%H:%M:%SZ')[-1],
+        'featureType': featureType,
+        'date_created': pd.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ"),
+        'platform_code': SHIP_CODE,
+        'vessel_name': ship_callsign_ls[SHIP_CODE],
+        'geospatial_lat_min': df.LATITUDE.dropna().min(),
+        'geospatial_lat_max': df.LATITUDE.dropna().max(),
+        'geospatial_lon_min': df.LONGITUDE.dropna().min(),
+        'geospatial_lon_max': df.LONGITUDE.dropna().max(),
+        'time_period': time_period,
+        'history': "File created %s" % pd.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+    })
+
+    nc_filename = 'IMOS_SOOP-TMV_TSUB_{time_start}_{vessel_code}_FV0{product_number}_{product_type}-{product_code}_END-{time_end}.nc'.format(
+        time_start=df.index.strftime('%Y%m%dT%H%M%SZ')[0],
+        time_end=df.index.strftime('%Y%m%dT%H%M%SZ')[-1],
+        vessel_code=SHIP_CODE,
+        product_number=0,
+        product_type=product_type,
+        product_code=product_code)
+
+    netcdf_path = os.path.join(output_dir, nc_filename)
+    template.to_netcdf(netcdf_path)
+    return netcdf_path
+
+
+class SoopTmvNrtHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(SoopTmvNrtHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc', '.log']
+
+    def preprocess(self):
+        f = self.file_collection[0]
+        if f.extension == '.nc':
+            # case where we repush an existing netcdf file
+            f.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+            f.dest_path = self.dest_path(f.src_path)
+
+        elif f.extension == '.log':
+            # case to create netcdf files from log files
+            f.publish_type = PipelineFilePublishType.NO_ACTION
+
+            log_path = self.file_collection[0].src_path
+            netcdf_filepath = netcdf_writer(log_path, self.temp_dir)
+
+            nc_file = PipelineFile(netcdf_filepath, dest_path=self.dest_path(netcdf_filepath))
+            nc_file.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+            self.file_collection.add(nc_file)
+
+    def dest_path(self, filepath):
+        ship_callsign_ls = ship_callsign_list()
```

Comment:
No point running this every time in the dest_path since it is only ever retrieving a static valu

I'd simplify it by storing the output of ship_callsign_ls[SHIP_CODE] in a instance attribute, and just having dest_path access that. Only "per file" decisions should be made in dest_path to keep it lean and efficient.